### ISSUE TITLE
BUG: PHPUnit test is failing in Moodle 3.11 and 4.0

### ISSUE THREAD

**Issue by michael-milette** [iarenaza/moodle-filter_multilang2#32]
When I run the unit test on multilang2, I get the following error:

    Class 'filter_test' could not be found in '/filter/multilang2/tests/filter_test.php'.

I suspect this is due to changes in PHPUnit 9.5.

Any chance you could fix this please?

Best regards,

Michael

**Comment by michael-milette** [comment]
Hi @iarenaza 

I am using PHPUnit 9.5 with PHP 7.4.25 and the latest release of moodle-local_codechecker. I tried the command line you used and got similar results to yours. I think it is a difference in the command line that I use:

    vendor/bin/phpunit filter/multilang2/tests/filter_test.php

The result is:

    Moodle 4.0.1+ (Build: 20220603), 6bafbcc92d6a7feb3e57d08eec4a8e10a1864196
    Php: 7.4.25, mariadb: 10.4.6
    Class 'filter_test' could not be found in '/filter/multilang2/tests/filter_test.php'.

The problem is that the PHPUnit requirements for class and file naming changed starting with Moodle 3.11 and up. For more information, see https://docs.moodle.org/dev/PHPUnit_integration#Class_and_file_naming_rules

On a side note, in order to avoid a code precheck phpcs warnings on Moodle.org, I recommend you also look into @covers. See:
https://phpunit.readthedocs.io/en/9.5/annotations.html#covers

Hope you find this information helpful.

Best regards,

Michael Milette

**Comment by michael-milette** [comment]
Hi @iarenaza 

FYI: Without the correction, you will get the following warning messages in the section PHP coding style problems in the precheck next time you upload try to load the plugin to Moodle.org. Note: This is just a simulation of the warning messages (I substituted filtercodes for multilang2 in the messages):

    PHPUnit testcase name "filter_multilang2_testcase" does not match file name "filter_test".
    
    PHPUnit class "filter_multilang2_testcase" does not have any namespace. It is recommended to add it to the "filter_multilang2" namespace, using more levels if needed, in order to match the code being tested.
    
    Test method test_filter_multilang2() is missing any coverage information, own or at class level.

I had similar issues with my FilterCodes plugin whose test file was configured similar to yours. If you are interested, visit https://github.com/michael-milette/moodle-filter_filtercodes/blob/master/tests/filter_test.php to see how I resolved it.

The version of FilterCodes on Moodle.org still generated one warning but that is because it does not include the @covers which is included in the version on GitHub.

I am a big fan of your plugin and just want to help you out. Thank you for all your efforts.

Michael
